sentry-core/src/main/java/io/sentry/core/SentryExecutorService.java

```diff
@@ -0,0 +1,42 @@
+package io.sentry.core;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.TestOnly;
+
+final class SentryExecutorService implements ISentryExecutorService {
+
+  private final @NotNull ExecutorService executorService;
+
+  @TestOnly
+  SentryExecutorService(final @NotNull ExecutorService executorService) {
+    this.executorService = executorService;
+  }
+
+  SentryExecutorService() {
+    this(Executors.newSingleThreadExecutor());
+  }
+
+  @Override
+  public Future<?> submit(final @NotNull Runnable runnable) {
+    return executorService.submit(runnable);
+  }
+
+  @Override
+  public void close(final long timeoutMillis) {
+    synchronized (executorService) {
+      if (!executorService.isShutdown()) {
+        executorService.shutdown();
+        try {
+          executorService.awaitTermination(timeoutMillis, TimeUnit.MILLISECONDS);
+        } catch (InterruptedException e) {
+          executorService.shutdownNow();
```

Comment:
Could this throw and we never call `interrupt` below?

Reply:
no based on the docs https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ExecutorService.html